tyr/clusters/mongo.py

```diff
@@ -54,6 +56,26 @@ def provision(self):
 
             zones += zones
 
+        subnets = []
+
+        if self.vpc_id:
+            
+            self.log.info("VPC ID provided. Checking for subnets.")
+
+            conn = boto.connect_vpc()
+            all_subnets = conn.get_all_subnets()
+
+            subnets = [subnet.id for subnet in all_subnets
```

Comment:
So while I can definitely appreciate the flexibility that dynamically populating the list of subnets brings us, I have a slight fear of this coming back to bite us. So a couple notes:

- we only use the availability zones `c`, `d`, and `e`
- a VPC can have multiple subnets using the same availability zone

So we would run into an issue if a new subnet was created. The new subnet might just be created for testing something or to isolate some instances in a separate subnet, but all of a sudden instances spun up with `MongoCluster` would automatically be placed inside the subnet.

If an experimental subnet were created for zone `a` or `b`, then what happened would be pretty apparent in the logs - you'd see an instance with a name like `p-monolith-mongo-use1a-01.app.hudl.com`. But what if the new subnet was created in zones `c`, `d`, or `e`? You'd see what looks like an expected instance name and might not realize what had happened if

- you don't spot the different subnet ID in the logs when the instance is being provisioned
- you don't spot the private IP address being placed in a new CIDR block

So basically, I'm worried about someone creating a new subnet which may not be intended for general use, `MongoCluster` automatically electing to place nodes there, and no one realizing what's happened.

It's not as flexible, but I think I might just lean towards hard coding it and then updating the list when we create a new subnet which is intended for use. Instead, or in addition, you could have an argument `subnet_ids` in addition to `vpc_id` so that developers can customize the subnets being used if they need to. If the value is `None`, use the hardcoded values, if not, use the list provided.